netcfgbu/connectors/basic.py

```diff
@@ -333,38 +324,49 @@ async def login(self):
             reason=f"No valid username/password, attempted {len(self.creds)} credentials."
         )
 
-    async def close(self):
+    async def close(self) -> None:
+        """Closes the SSH connection."""
         self.conn.close()
         await self.conn.wait_closed()
-        self.log.info(f"CLOSED: {self.name}")
+        self.log.info("CLOSED: %s", self.name)
 
     # -------------------------------------------------------------------------
     #
     #                                    Helpers
     #
     # -------------------------------------------------------------------------
 
-    async def read_until_prompt(self):
+    async def read_until_prompt(self) -> bytes:
+        """Reads from the SSH process until the prompt is detected.
+
+        Returns:
+            The output read until the prompt is found as a byte string.
+        """
         output = b""
         while True:
+            self.log.debug("%s - %s", self.name, output)
             output += await self.process.stdout.read(io.DEFAULT_BUFFER_SIZE)
```

Comment:
**Suggestion:** The `read_until_prompt` method should include a timeout parameter to avoid potential infinite loops or hanging if the prompt is never reached. [enhancement, importance: 8]
```suggestion
output += await asyncio.wait_for(self.process.stdout.read(io.DEFAULT_BUFFER_SIZE), timeout=30)
```